templates/components/vectordbs/python/pg/vectordb.py

```diff
@@ -5,26 +5,33 @@
 PGVECTOR_SCHEMA = "public"
 PGVECTOR_TABLE = "llamaindex_embedding"
 
+vector_store: PGVectorStore = None
+
 
 def get_vector_store():
```

Comment:
sec insights is using a TTL cache for storage context (see https://github.com/run-llama/sec-insights/blob/94ec3507ca5ea8f67c52d82ef69d18953b2766ad/backend/app/chat/engine.py#L121)

1. can we also use that for vector_store? sounds better than having a global vector_store
2. if we don't create a vector store per request, the vector_store must be thread safe. is it?
3. does it make sense to use this ttl cache also for the whole storage context?

Reply:
1. Yes, we can use the cache, just simple cache is enough. 
2. It must be thread safe but I think in fastapi app each request is already handling by a uvicorn worker, it's working now so it should be working with the update.
3. We’re not using all of the storage context store, but it’s worth applying it to other vector stores too.
How about using a singleton object since it's easier to handle (lazy init,..).